### ISSUE TITLE
When redis CPU is maxed, there are fewer number of writes through hocuspocus server

### ISSUE THREAD

**Issue by ralphiee22** [ueberdosis/hocuspocus#588]
**Description**
There appears to be a correlation between the redis cpu engine being maxed out, and the lower number of writes the hocuspocus server executes.
I have not pinned down exactly what the cause is, but I do have a theory:
- Redis can not expire/let go of locks, and this throws in the `onStoreDocument` hook of `extension-redis`, causing subsequent hooks/writes to not go through (specifically the on store hooks we have set up to write to the db)

**Expected behavior**
Writes will still go through regardless of redis cpu state.

**Screenshot, video, or GIF**
![cpu vs writes](https://user-images.githubusercontent.com/6361732/234111411-b592eb26-72c4-48c3-8e3c-9048479ebafa.png)

**Environment?**
 - Hocuspocus version: 
```
"@hocuspocus/extension-redis": "1.0.2",
"@hocuspocus/server": "1.0.2",
```

**Additional context**
Is there anything we can do to get around the redis extension locking (and preventing writes)?
Or to at least add some logging in this extension to see if we see an increased amount of errors during high cpu times?

**Comment by janthurau** [comment]
What do you mean by `writes the hocuspocus server executes` exactly? You mean writes to the database (i.e. onStoreDocument executions), or do you mean writes to the yDoc? I'm guessing the first, but just to make sure.

I wanted to remove the redisLock anyway (see #453), but I haven't received a lot of feedback there so I stopped working. If you wanna try it out, I could release a pre version for this to see if this solves your issue. The locking would need to be handled by your database / whatever you are doing in onStoreDocument though (I guess that is expected though).

**Comment by ralphiee22** [comment]
> You mean writes to the database (i.e. onStoreDocument executions)

Yes I mean writes to the database.

If you can release a pre-version that would be great so we can try it out (and control the locking on our side).

I also brought up this issue because it seems a design flaw that a maxed redis could potentially block out db writes (onStoreDocument), leading to data loss.

**Comment by janthurau** [comment]
@ralphiee22I have just released v2.1.0-alpha.0 which adds compatibility to ioredis^5.0.4 and removes redlock. Please let me know if that breaks something (or, in your case, fixes your issue :))

Here's the related PR: https://github.com/ueberdosis/hocuspocus/pull/453

**Comment by janthurau** [comment]
@ralphiee22 as this is an edge case (probably the primary issue is redis being at 100% CPU usage), I'll close this for now. 

Let me know if the release pre version fixes your issue, otherwise I'll just queue it for the next major release.
